test/integration/http_integration.cc

```diff
@@ -1063,6 +1064,136 @@ void HttpIntegrationTest::testHittingEncoderFilterLimit() {
   EXPECT_STREQ("500", response->headers().Status()->value().c_str());
 }
 
+// Verifies metadata can be sent at different locations of the responses.
+void HttpIntegrationTest::testEnvoyProxyMetadataInResponse() {
+  initialize();
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+
+  // Sends the first request.
+  auto response =
+      codec_client_->makeRequestWithBody(Http::TestHeaderMapImpl{{":method", "GET"},
+                                                                 {":path", "/test/long/url"},
+                                                                 {":scheme", "http"},
+                                                                 {":authority", "host"}},
```

Comment:
Might as well declare this HeaderMapImpl once at the top and use it in all 3 places.  Alternately can you use default_request_headers_?

Reply:
Good idea! Fixed.